### PR TITLE
Initial Event Processor

### PR DESCRIPTION
**Pull request by peterschwarz**
This event processor is configured to receive the appropriate events from a Sawtooth Validator, which can be offloaded into a reporting db (to be added in a subsequent PR).

### CODE REVIEW

File: daemon/src/event/mod.rs [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        let event_handler = event_handlers
+            .iter()
+            .find(|handler| handler.event_type() == event.get_event_type());
+
+        if let Some(event_handler) = event_handler {
+            if let Err(err) = event_handler.handle_event(&event) {
+                error!("Unable to handle event {}: {}", event.get_event_type(), err);
+            }
+        } else {
+            warn!("Unable to handl event {}", event.get_event_type());
```
Comment (jsmitchell) [inline daemon/src/event/mod.rs]: s/handl/handle/

File: daemon/src/main.rs [hunk 2]
```diff
@@ -53,18 +58,45 @@ fn run() -> Result<(), DaemonError> {
...
+    )
+    .map_err(|err| DaemonError::EventProcessorError(Box::new(err)))?;
+
+    let (event_processor_shutdown_handle, event_processor_join_handle) =
+        evt_processor.take_shutdown_controls();
 
+    let ctrlc_triggered = AtomicBool::new(false);
     ctrlc::set_handler(move || {
+        if ctrlc_triggered.load(Ordering::SeqCst) {
+            process::exit(0);
```
Comment (vaporos) [inline daemon/src/main.rs]: This should probably be process::exit(1). This should maybe print something out to stderr prior to exiting like "Aborting due to multiple Ctrl-C events...".

File: daemon/src/rest_api/mod.rs [hunk 3]
```diff
@@ -55,21 +54,31 @@ pub fn run(
...
+                .start();
 
             tx.send(addr).map_err(|err| {
                 RestApiError::StartUpError(format!("Unable to send Server Addr: {}", err))
             })?;
 
-            Ok(sys.run())
+            sys.run();
+
+            info!("Rest API terminiating");
```
Comment (vaporos) [inline daemon/src/rest_api/mod.rs]: Typo

File: docker-compose.yaml [hunk 4]
```diff
@@ -97,11 +97,21 @@ services:
...
       context: .
       dockerfile: daemon/Dockerfile
     expose:
       - 8080
     ports:
       - '8080:8080'
-    entrypoint: "/gridd -v -b gridd:8080"
+    entrypoint: /gridd -vv -b gridd:8080 -C tcp://validator:4004
+
+  sawtooth-shell:
```
Comment (vaporos) [inline docker-compose.yaml]: We are probably going to want to create our own grid shell instead and install sawtooth and grid CLIs into it.

Comment (peterschwarz) [reply docker-compose.yaml]: I'll leave that to a future PR.

**Pull request merged**